tests/server/wildfly/src/test/java/org/hawkular/apm/tests/wildfly/zipkin/SpanAnalyticsITest.java

```diff
@@ -0,0 +1,253 @@
+        urlRootBinAnn.setValue("http://localhost/root");
+        urlRootBinAnn.setType(AnnotationType.STRING);
+
+        Span rootSpan = new Span(Arrays.asList(urlRootBinAnn),
+                SpanServiceITest.serverAnnotations(
+                        SpanServiceITest.millisToMicroSeconds(1), SpanServiceITest.millisToMicroSeconds(30)));
+        rootSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(1L));
+        rootSpan.setDuration(SpanServiceITest.calculateDuration(1L, 30L));
+        rootSpan.setTraceId("root");
+        rootSpan.setId("root");
+        rootSpan.setName("get");
+
+        BinaryAnnotation urlServiceABinAnn = new BinaryAnnotation();
+        urlServiceABinAnn.setKey(Constants.ZIPKIN_BIN_ANNOTATION_HTTP_URL);
+        urlServiceABinAnn.setValue("http://localhost/serviceA");
+
+        Span clientSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.clientAnnotation(SpanServiceITest.millisToMicroSeconds(5),
+                        SpanServiceITest.millisToMicroSeconds(15)));
+        clientSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(5));
+        clientSpan.setDuration(SpanServiceITest.calculateDuration(5L, 15L));
+        clientSpan.setTraceId(rootSpan.getTraceId());
+        clientSpan.setId("2");
+        clientSpan.setParentId("root");
+        clientSpan.setName("post");
+
+        Span serverSpan = new Span(Arrays.asList(urlServiceABinAnn),
+                SpanServiceITest.serverAnnotations(
+                        SpanServiceITest.millisToMicroSeconds(7), SpanServiceITest.millisToMicroSeconds(10)));
+        serverSpan.setTimestamp(SpanServiceITest.millisToMicroSeconds(7));
+        serverSpan.setDuration(SpanServiceITest.calculateDuration(7L, 10L));
+        serverSpan.setTraceId(rootSpan.getTraceId());
+        serverSpan.setId("2");
+        serverSpan.setParentId("root");
+        serverSpan.setName("post");
+
+        post(Server.Zipkin, "/spans", null, Arrays.asList(rootSpan));
+        post(Server.Zipkin, "/spans", null, Arrays.asList(clientSpan));
+        post(Server.Zipkin, "/spans", null, Arrays.asList(serverSpan));
+
+        Wait.until(() -> {
+            List<CompletionTime> traceCompletionTimes = analyticsService.getTraceCompletionTimes(null, requestCriteria());
+            return traceCompletionTimes == null || traceCompletionTimes.size() != 1 ? false : true;
+        }, 30, TimeUnit.SECONDS);
+
+        List<CompletionTime> traceCompletionTimes = analyticsService.getTraceCompletionTimes(null, requestCriteria());
+        Assert.assertEquals(1, traceCompletionTimes.size());
+
+        CompletionTime completionTime = traceCompletionTimes.get(0);
+        Assert.assertEquals(rootSpan.getDuration(),
+                TimeUnit.MICROSECONDS.convert(completionTime.getDuration(), TimeUnit.MILLISECONDS));
+        Assert.assertEquals("/root", completionTime.getUri());
+        Assert.assertEquals("root", completionTime.getId());
+    }
+
```

Comment:
Could you also add an example where the completion time is longer than the root span duration - i.e. the async case.